lib/meadow/data/schemas/validations.ex

```diff
@@ -0,0 +1,24 @@
+defmodule Meadow.Data.Schemas.Validations do
+  @moduledoc """
+  This module provides custom changeset functions and
+  validations
+  """
+
+  @doc """
+  If a `cast_embed()` will result in a `nil` value (either on create or
+  update), set it to an empty embedded struct instead
+  """
```

Comment:
Would this be a good module/function to try out a DocTest? I'd been looking for somewhere to do this - this seems like a good candidate.